src/IECore/FileNameParameter.cpp

```diff
@@ -87,7 +87,7 @@ bool FileNameParameter::valueValid( const Object *value, std::string *reason ) c
 	// extensions check
 	if( extensions().size() )
 	{
-		string ext = boost::filesystem::extension(boost::filesystem::path( s->readable()));
+		string ext = boost::filesystem::path(boost::filesystem::path( s->readable())).extension().string();
```

Comment:
I think there is one more `filesystem::path()` call here than we need. Could just be `boost::filesystem::path( s->readable() ).extension().string()`.

Reply:
yes. it is fixed in the next commit.